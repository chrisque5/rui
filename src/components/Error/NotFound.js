import { useHistory } from 'react-router-dom';
import logo from '../../img/tpicap.png';

const NotFound = () => {
  const history = useHistory();
  return (
    <div className="error-page">
      <div className="flex-container">
        <div className="flex-item">
          <img src={logo} alt="Logo" className="logo" />

          <h3 className="title">Deal Management System</h3>
          <h4 className="subtitle">404 - Not Found</h4>
          <p>
            Page &quot;
            {history.location.pathname}
            &quot; not found.
          </p>
          <button type="button" className="linkbutton" onClick={history.goBack}>Go back</button>
        </div>
      </div>
    </div>
  );
};
export default NotFound;
