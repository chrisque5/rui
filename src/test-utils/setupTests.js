// The below part of the code is to fix the issue
// TypeError: window.matchMedia is not a function
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation((query) => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn(),
  })),
});
