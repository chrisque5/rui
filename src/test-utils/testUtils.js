/* eslint-disable import/no-extraneous-dependencies, react/prop-types */
import { render } from '@testing-library/react';
import { Provider } from 'react-redux';
import configureMockStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import { Form } from 'antd';

const middlewares = [thunk];
const renderWithForm = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}><Form>{children}</Form></Provider>;

  return render(ui, { wrapper, ...renderOptions });
};

const renderWithProvider = (
  ui,
  {
    initialState,
    store = configureMockStore(middlewares)(initialState),
    ...renderOptions
  } = {},
) => {
  const wrapper = ({ children }) => <Provider store={store}>{children}</Provider>;

  return render(ui, { wrapper, ...renderOptions });
};
// re-export everything
export * from '@testing-library/react';
// override render method
export { renderWithForm, renderWithProvider };
