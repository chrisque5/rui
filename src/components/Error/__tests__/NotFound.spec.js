import { render } from 'test-utils/testUtils';
import NotFound from '../NotFound';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useHistory: () => ({
    location: {
      pathname: 'localhost:3000/unknown/path',
    },
  }),
}));

describe('<NotFound>', () => {
  test('should match the snapshot', () => {
    const { container } = render(<NotFound />);
    expect(container).toMatchSnapshot();
  });
});
