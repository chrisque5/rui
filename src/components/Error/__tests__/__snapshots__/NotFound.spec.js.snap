// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NotFound> should match the snapshot 1`] = `
<div>
  <div
    class="error-page"
  >
    <div
      class="flex-container"
    >
      <div
        class="flex-item"
      >
        <img
          alt="Logo"
          class="logo"
          src="tpicap.png"
        />
        <h3
          class="title"
        >
          Deal Management System
        </h3>
        <h4
          class="subtitle"
        >
          404 - Not Found
        </h4>
        <p>
          Page "
          localhost:3000/unknown/path
          " not found.
        </p>
        <button
          class="linkbutton"
          type="button"
        >
          Go back
        </button>
      </div>
    </div>
  </div>
</div>
`;
